sdk/python/kfp/dsl/_metadata.py

```diff
@@ -46,24 +46,36 @@ def to_dict_or_str(self):
       return {self.name: self.properties}
 
   @staticmethod
-  def from_dict_or_str(json):
+  def from_dict_or_str(payload):
+    '''from_dict_or_str accepts a payload object and returns a TypeMeta instance
+     Args:
+       payload (str/dict): the payload could be a str or a dict
+    '''
+
     type_meta = TypeMeta()
-    if isinstance(json, str) and '{' in json:
-      import ast
-      json = ast.literal_eval(json)
-    if isinstance(json, dict):
-      if not _check_valid_type_dict(json):
-        raise ValueError(json + ' is not a valid type string')
-      type_meta.name, type_meta.properties = list(json.items())[0]
+    if isinstance(payload, dict):
+      if not _check_valid_type_dict(payload):
+        raise ValueError(payload + ' is not a valid type string')
+      type_meta.name, type_meta.properties = list(payload.items())[0]
       # Convert possible OrderedDict to dict
       type_meta.properties = dict(type_meta.properties)
-    elif isinstance(json, str):
-      type_meta.name = json
+    elif isinstance(payload, str):
```

Comment:
Do you need to have another else to throw if the payload is not dict or str? I am worry that the ast might convert to unexpected types and we will lose the debug information here.

Reply:
Good idea. Done.